gradio/components/login_button.py

```diff
@@ -86,21 +90,29 @@ def _check_login_status(self, request: Request) -> LoginButton:
             return LoginButton(value=self.value, interactive=True)
         else:
             username = session["oauth_info"]["userinfo"]["preferred_username"]
-            signed_in_text = self.signed_in_value.format(username)
-            return LoginButton(signed_in_text, interactive=False)
+            logout_text = self.logout_value.format(username)
+            return LoginButton(logout_text, interactive=True)
 
 
-# JS code to redirects to /login/huggingface if user is not logged in.
-# If the app is opened in an iframe, open the login page in a new tab.
-# Otherwise, redirects locally. Taken from https://stackoverflow.com/a/61596084.
-_js_open_if_not_logged_in = """
-(buttonValue) => {
-    if (!buttonValue.includes("Signed in")) {
-        if ( window !== window.parent ) {
-            window.open('/login/huggingface', '_blank');
-        } else {
-            window.location.assign('/login/huggingface');
+def _js_handle_redirect(button_value: str) -> str:
+    # JS code to redirects to /login/huggingface if user is not logged in.
+    # If the app is opened in an iframe, open the login page in a new tab.
+    # Otherwise, redirects locally. Taken from https://stackoverflow.com/a/61596084.
+    # If user is logged in, redirect to logout page (always in-place).
+    return (
+        """
+        (buttonValue) => {
+            if (buttonValue.includes('"""
+        + button_value
```

Comment:
This opens an avenue for JavaScript code injection (consider `buttonValue` being `' + alert("evil") + '`). Use `json.dumps()` to include a string in JavaScript.